packages/backend/src/clients/Google/GoogleDriveClient.ts

```diff
@@ -306,12 +296,46 @@ export class GoogleDriveClient {
             }),
         );
 
+        await this.appendCsvToSheet(
+            refreshToken,
+            fileId,
+            [csvHeader, ...values],
+            tabName,
+        );
+    }
+
+    async appendCsvToSheet(
+        refreshToken: string,
+        fileId: string,
+
+        results: string[][],
+        tabName?: string,
+    ) {
+        if (!this.isEnabled) {
+            throw new Error('Google Drive is not enabled');
+        }
+
+        const auth = await this.getCredentials(refreshToken);
+        const sheets = google.sheets({ version: 'v4', auth });
+
+        // Clear first sheet before writting
+        await GoogleDriveClient.clearTabName(sheets, fileId, tabName);
+
+        if (results.length === 0) {
+            Logger.info('No data to write to the sheet');
+            return;
+        }
```

Comment:
should this happen before checking credentials and clearing the tab?

Reply:
done